kustomize/components/spanner/README.md

```diff
@@ -0,0 +1,87 @@
+# Integrate Online Boutique with Spanner
+
+By default the `cartservice` app is serializing the data in an in-cluster Redis database. Using a database outside your GKE cluster could bring more resiliency and more security with a managed service like Google Cloud Spanner.
+
+## Provision a Spanner database
+
+To provision a Spanner database you can follow the following instructions:
+```bash
+gcloud services enable spanner.googleapis.com
+
+SPANNER_REGION_CONFIG="<your-spanner-region-config-name>" # e.g. "regional-us-east5"
+SPANNER_INSTANCE_NAME=onlineboutique
+SPANNER_DATABASE_NAME=carts
+
+gcloud spanner instances create ${SPANNER_INSTANCE_NAME} \
+    --config ${SPANNER_REGION_CONFIG}
+
+gcloud spanner databases create ${SPANNER_DATABASE_NAME} \
+    --instance ${SPANNER_INSTANCE_NAME} \
+    --database-dialect GOOGLE_STANDARD_SQL \
+    --ddl-file ./src/cartservice/ddl/CartItems.ddl
+```
+_Note: with latest version of `gcloud` you can create a free Spanner instance by leveraging the `--instance-type free-instance` parameter._
+
+## Grant the `cartservice`'s service account access to the Spanner database
+
+Important note:
+- Your GKE cluster should have [Workload Identity enabled](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable).
+
+As a good practice, let's create a dedicated least privilege Google Service Account to allow the `cartservice` to communicate with the Spanner database:
+```bash
+PROJECT_ID=<your-project-id>
+SPANNER_DB_USER_GSA_NAME=spanner-db-user-sa
+SPANNER_DB_USER_GSA_ID=${SPANNER_DB_USER_GSA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com
+ONLINEBOUTIQUE_NAMESPACE=default
+CARTSERVICE_KSA_NAME=cartservice
+
+gcloud iam service-accounts create ${SPANNER_DB_USER_GSA_NAME} \
+    --display-name=${SPANNER_DB_USER_GSA_NAME}
+gcloud spanner instances add-iam-policy-binding ${SPANNER_INSTANCE_NAME} \
+    --member "serviceAccount:${SPANNER_DB_USER_GSA_ID}" \
+    --role roles/spanner.databaseUser
+
+gcloud iam service-accounts add-iam-policy-binding ${SPANNER_DB_USER_GSA_ID} \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[${ONLINEBOUTIQUE_NAMESPACE}/${CARTSERVICE_KSA_NAME}]" \
+    --role roles/iam.workloadIdentityUser
+```
+
+## Deploy Online Boutique connected to a Spanner database
+
+To automate the deployment of Online Boutique integrated with Spanner you can leverage the following variation with [Kustomize](../..).
+
+From the `kustomize/` folder at the root level of this repository, execute these commands:
+```bash
+kustomize edit add component components/service-accounts
```

Comment:
We are creating 1 KSA per app, with that we could just assigned the Spanner User GSA specifically to the `cartservice` KSA.